core/pkg/sync/file/filepath_sync.go

```diff
@@ -41,14 +59,27 @@ func (fs *Sync) ReSync(ctx context.Context, dataSync chan<- sync.DataSync) error
 	return nil
 }
 
-func (fs *Sync) Init(_ context.Context) error {
+func (fs *Sync) Init(ctx context.Context) error {
 	fs.Logger.Info("Starting filepath sync notifier")
-	w, err := fsnotify.NewWatcher()
-	if err != nil {
-		return fmt.Errorf("error creating filepath watcher: %w", err)
+
+	switch fs.watchType {
+	case FSNOTIFY, "":
+		w, err := NewFSNotifyWatcher()
+		if err != nil {
+			return fmt.Errorf("error creating fsnotify watcher: %w", err)
+		}
+		fs.watcher = w
+	case FILEINFO:
+		w := NewFileInfoWatcher()
+		// start the timer
+		// TODO: wire up configuration for the fileinfo sync duration
+		w.Run(ctx, 1*time.Second)
```

Comment:
`w.Run(...)` should be internal to the `fileInfoWatcher`. You can start the internal routine with `NewFileInfoWatcher()` constructor.

Reply:
thank you for reminding  me, I meant to add a bullet about this.  the goroutine lifecycle for the timer is going to require a bit of rework; as written, the run method is blocking (which breaks init in its current form), and I wasnt sure how we felt about moving run() into the constructor.  At my dayjobs its usually a no-no for the constructor to do anything but construct

Reply:
@djosephsen Sure noted. For flagd, I am fine to start the background go routine through the constructor. 

Besides, we need some error tolerance within the Run method. Because right now error will end the routine and there's no checks for the `erroGroup` as there's no wait. In my view we could even avoid the `erroGroup` and rely on panic - recover for extreme cases. But I will leave the decision to you :)